### DISCUSSION TITLE
Nuclei Stats

### DISCUSSION THREAD

**Discussion by rxerium** [projectdiscovery/nuclei#5760] [General]
Hi team, 

I understand when viewing Nuclei stats (`-stats`) in terminal a lot of the values are incorrect. Can you confirm:
1. Which values are incorrect? (i,e, RPS, requests etc..)
2. How incorrect are they?
3. Are there any upcoming fixes to resolve this issue?

Many thanks,

Rishi

**Comment by rxerium** [comment]
Hey team, just following up here :)

**Comment by GeorginaReeder** [comment]
Hey @rxerium ! Thanks for flagging :)

Tagging @dwisiswant0 to take a look, thanks for your patience!

**Comment by tarunKoyalwar** [comment]
Hey @rxerium,

Partially Incorrect Stats:

- Scan percentage
- Requests

Currently, stats don't account for conditions in flow and dynamic extractors. This omits a key metric in terms of request count (we have requests_sent and total_requests, but we need to support skipped_requests). This directly affects scan percentage, as there will always be a difference between actual requests sent and total estimated requests.

The known solution is to add another metric—skipped_requests—and consider it when calculating percentages. This requires a refactor at the template and protocol level, for which we've created an issue.

Regarding EA, there are known pending refactors of certain components that we're trying to cover with planned refactors. We'll try to prioritize this, but generally, we follow this approach: template bugs > engine bugs > new features ~ refactor )

**Reply by rxerium** [reply]
Thanks for clarifying that!
